### PR TITLE
Update configuration-reference.md

### PR DESCRIPTION
**Pull request by leoj3n**
Add more info about draft pages, and other default values.

### CODE REVIEW

File: src/pages/en/reference/configuration-reference.md [hunk 1]
```diff
@@ -63,10 +75,36 @@ The `renderers` option defines the framework renderers to be used by Astro.
 
 The `buildOptions` option configures how a site is built, including its base URL (`buildOptions.site`), whether it includes a sitemap (`buildOptions.sitemap`), whether markdown draft pages should be included in the build (`buildOptions.drafts`), and whether its pages should be files (`path.html`) or directories (`path/index.html`) (`buildOptions.pageUrlFormat`).
 
+**Defaults**:
+
+- `buildOptions.site`: Used to generate sitemaps and canonical URLs.
```
Comment (jonathantneal) [inline src/pages/en/reference/configuration-reference.md:80]: I like this. Would this voicing update be acceptable?
```suggestion
- `buildOptions.site`: Public [origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) used to generate sitemaps and canonical URLs.
```

File: src/pages/en/reference/configuration-reference.md [hunk 2]
```diff
@@ -63,10 +75,36 @@ The `renderers` option defines the framework renderers to be used by Astro.
...
+  - Your public domain, e.g.: `https://my-site.dev/`.
+- `buildOptions.sitemap`: Generate an automatically-generated sitemap for your build.
```
Comment (jonathantneal) [inline src/pages/en/reference/configuration-reference.md:82]: Voicing suggestion.
```suggestion
- `buildOptions.sitemap`: Whether to automatically generate a sitemap.
```

File: src/pages/en/reference/configuration-reference.md [hunk 3]
```diff
@@ -63,10 +75,36 @@ The `renderers` option defines the framework renderers to be used by Astro.
...
+  - Either `true` or `false`.
+  - Default: `true`.
+- `buildOptions.pageUrlFormat`: Control the output file URL format of each page.
```
Comment (jonathantneal) [inline src/pages/en/reference/configuration-reference.md:85]:
```suggestion
- `buildOptions.pageUrlFormat`: Determines how files built from pages are written.
```

File: src/pages/en/reference/configuration-reference.md [hunk 4]
```diff
@@ -63,10 +75,36 @@ The `renderers` option defines the framework renderers to be used by Astro.
...
+  - Either `file` (ex: "/foo.html") or `directory` (ex: "/foo/index.html").
+  - Default: `'directory'`.
+- `buildOptions.drafts`: Control if markdown draft pages should be included in the build.
```
Comment (jonathantneal) [inline src/pages/en/reference/configuration-reference.md:88]:
```suggestion
- `buildOptions.drafts`: Determines whether markdown draft pages are included in the build.
```

File: src/pages/en/reference/configuration-reference.md [hunk 5]
```diff
@@ -63,10 +75,36 @@ The `renderers` option defines the framework renderers to be used by Astro.
...
+  - Either `true` or `false`.
+  - Default: `false`.
+
+Read more about [markdown draft pages][markdown-draft-pages].
```
Comment (jonathantneal) [inline src/pages/en/reference/configuration-reference.md:92]: ❤️

**Review by jonathantneal — CHANGES_REQUESTED** [review]
This is another helpful addition. Thank you, @leoj3n.

I’ve left some suggestions to change how a few lines are written for your consideration. I’m happy to work with you on these, if you feel there’s more to be done. Would you please take a look?

**Comment by FredKSchott** [comment]
Thanks @leoj3n! This is a nice improvement over what we have 👍 

If you're interested in contributing more, the rough plan is to start auto-generating a lot of this page contents from the codebase itself. I'd love your help in https://github.com/withastro/docs/pull/172 if you're interested.

**Pull request merged**